### PR TITLE
[Enhancement] Imporve predicate statistics cost

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

1.  EnforceAndCostTask remove call PredicateStatisticsCalculator
    we will choose best statistics in statistics deriver task, here is unnecessary

2. Rmove confidenceStatistics from group expression, it's useless now

Todo:
we will choose the fewest output rows statistics now, may be we need update statistics choose algorithm, like:
* the fewest predicate statistics ?
* the avg(output rows) statistics ?
* ... and so on

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
http://39.101.206.100:8090/job/starrocks_benchmark/878/Pre_20optimized_20compare_20report/

**Review by kangkaisen — DISMISSED** [review]

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 17 (94.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/QueryDumpAction.java|8|9|88.89%|[115]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ExpressionContext.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|4|4|100.00%|[]|

**Comment by Seaven** [comment]
the new perforcemance report
http://39.101.206.100:8090/job/starrocks_benchmark/907/Pre_20optimized_20compare_20report/

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 17 (94.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/QueryDumpAction.java|8|9|88.89%|[115]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/EnumeratePlan.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/ExpressionContext.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|4|4|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/EnumeratePlan.java [hunk 1]
```diff
@@ -120,9 +120,7 @@ public static OptExpression extractNthPlanImpl(Group group, PhysicalPropertySet
         OptExpression chooseExpression = OptExpression.create(chooseGroupExpression.getOp(), childPlans);
         // record childrenOutputProperties at optExpression, used for planFragment builder to determine join type
         chooseExpression.setRequiredProperties(childrenOutputProperties);
-        chooseExpression.setStatistics(group.hasConfidenceStatistic(outputProperty) ?
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/EnumeratePlan.java:123]: Stupid question: when using confidence statistics against another?

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
run starrocks_admit_test

**Pull request merged**